### PR TITLE
fix(static/metrics/instance): fix duplicate metrics registration panic when recreating the instance

### LINKED ISSUE #6548: 0.40.0: panic: duplicate metrics collector registration attempted
Coming from https://github.com/NixOS/nixpkgs/pull/292122, where our VM test fails with a similar looking panic from #6510 and #6511.

We start `grafana-agent` via `grafana-agent -config.expand-env -config.file <path to config file>` with the following config:

```yaml
integrations:
  agent:
    enabled: true
    scrape_integration: true
  node_exporter:
    enabled: true
metrics:
  global:
    scrape_interval: 5s
  wal_directory: ${STATE_DIRECTORY}
```

Resulting in the following panic.
Can provide full logs if needed.

```
panic: duplicate metrics collector registration attempted
goroutine 301 [running]:
github.com/prometheus/client_golang/prometheus.(*wrappingRegisterer).MustRegister(0xc002f581b0, {0xc001ec8eb0?, 0x1, 0x3?})
        github.com/prometheus/client_golang@v1.18.0/prometheus/wrap.go:104 +0x145
github.com/prometheus/prometheus/storage/remote.NewWriteHandler({0x89c5200?, 0xc00333bd60}, {0x8a15f10, 0xc002f581b0}, {0x7f8335158a78?, 0xc00333d0a0})
        github.com/prometheus/prometheus@v1.99.0/storage/remote/write_handler.go:55 +0x1ac
github.com/grafana/agent/pkg/metrics/instance.(*Instance).initialize(0xc0033bc1c0, {0x8a3a170, 0xc00333be00}, {0x8a15ee0, 0xc001c59f38}, 0xc0002e2500)
        github.com/grafana/agent/pkg/metrics/instance/instance.go:412 +0x1fa
github.com/grafana/agent/pkg/metrics/instance.(*Instance).Run(0xc0033bc1c0, {0x8a3a170, 0xc00333be00})
        github.com/grafana/agent/pkg/metrics/instance/instance.go:309 +0x328
github.com/grafana/agent/pkg/metrics/instance.(*BasicManager).runProcess(0xc0031d4ac0, {0x8a3a170, 0xc00333be00}, {0xc003335a40, 0x20}, {0x8a58d10, 0xc0033bc1c0})
        github.com/grafana/agent/pkg/metrics/instance/manager.go:265 +0xa3
github.com/grafana/agent/pkg/metrics/instance.(*BasicManager).spawnProcess.func1()
        github.com/grafana/agent/pkg/metrics/instance/manager.go:235 +0x53
created by github.com/grafana/agent/pkg/metrics/instance.(*BasicManager).spawnProcess in goroutine 1
        github.com/grafana/agent/pkg/metrics/instance/manager.go:234 +0x345
```

Version info is different from the official release binaries, because we compile from source in nixpkgs.

```bash
$ grafana-agent --version
agent, version 0.40.0 (branch: v0.40.0, revision: v0.40.0)
  build user:       nix
  build date:       1980-01-01T00:00:00Z
  go version:       go1.21.7
  platform:         linux/amd64
  tags:             builtinassets,nonetwork,nodocker,promtail_journal_enabled
```

Just to avoid confusion:
`ldflags` are set here: https://github.com/NixOS/nixpkgs/blob/bb6e397a6a0318140fb0ba36d7612f1c6704f879/pkgs/servers/monitoring/grafana-agent/default.nix#L34-L44
and build `tags` here: https://github.com/NixOS/nixpkgs/blob/bb6e397a6a0318140fb0ba36d7612f1c6704f879/pkgs/servers/monitoring/grafana-agent/default.nix#L48-L53

### PR DESCRIPTION
**Pull request by hainenber**
#### PR Description
Props to @rfratto for troubleshooting and guidance, I'm just implementing the idea here :D 

#### Which issue(s) this PR fixes

Fixes #6548 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Tests updated

### CODE REVIEW

File: internal/static/metrics/instance/instance_test.go [hunk 1]
```diff
@@ -278,7 +278,6 @@ func TestInstance_Recreate(t *testing.T) {
 
 	// Recreate the instance, no panic should happen.
 	require.NotPanics(t, func() {
-		inst, err := New(prometheus.NewRegistry(), cfg, walDir, logger)
```
Comment (hainenber) [inline internal/static/metrics/instance/instance_test.go:281]: This actually recreates a blank new Prom registry and the children write handler won't have any duplicate registration, thus no panic.

Removing it allows for reproduction of the panic

Comment (rfratto) [reply internal/static/metrics/instance/instance_test.go:281]: I don't think we should remove this line (the comment doesn't recreate the instance anymore), but rather pass in the same registry on both line 261 and here. Can we add it back and make that change?

**Review by rfratto — COMMENTED** [review]
Thanks! Once the comment about the test is resolved I'll merge and get a patch release out for this.

File: internal/static/metrics/instance/instance_test.go (hunk 1 above)
Comment (hainenber) [reply internal/static/metrics/instance/instance_test.go:281]: Re-added :D

Comment (hainenber) [reply internal/static/metrics/instance/instance_test.go:281]: And I can confirm that without the change in `instance.go`, that also reproduces the panic.

**Review by rfratto — APPROVED** [review]
Thanks again!

**Pull request merged**